### DISCUSSION TITLE
Handling embedded fonts in PDFs

### DISCUSSION THREAD

**Discussion by Woody1193** [py-pdf/pypdf#2618] [Q&A]
I have a pretty generic script designed to extract text from PDFs I receive. Recently, however, two PDFs failed this process. When I tried debugging it, none of the text `page.extract_text()` was returning matched what I could actually see in the document itself.

Digging deeper, I discovered three base fonts, which were also embedded:

```
{
  140012660527056: '/GEAKCA+Arial',
  140012660532816: '/GEAKDA+俵俽俹僑僔僢僋',
  140012660454864: '/GEAKBP+俵俽僑僔僢僋'
}
```

These fonts are different from what I've had to work with otherwise and I would assume that this is what resulted in the errors. However, 
 I'm unable to find any references to these fonts online and I'm not sure how to deal with this issue as I'm not particularly well-versed with the PDF standard. What steps should I take to debug this further and how could I resolve this issue?

**Comment by stefan6419846** [comment]
It is rather unlikely that we will be able to help you with this unless you provide a corresponding PDF file with an offending page for further analysis.

**Reply by Woody1193** [reply]
Sorry, I'd love to provide it but it's confidential so I'm afraid I cannot. However, I also tried selecting the text and copying it to notepad and noticed the same issue so I think it's a problem with my machine missing the font itself rather than an issue with the PDF's encoding.

**Reply by pubpub-zz** [reply]
Fonts are embedded in PDF. The issue is morelikely due to missing translation table (in '/ToUnicode' field of the font)

**Reply by pubpub-zz** [reply]
Isn't there a front page or index you could provide that could not contain confidential data?

**Reply by stefan6419846** [reply]
When copying fails, this either is some unexpectedly missing translation table as already pointed out, some limitation of the application (different application might handle this differently) or an intended obfuscation technique.

**Reply by ohsix** [reply]
font substitution schemes are a well-known-ish obfuscation for pdfs, as a side effect, it makes them unsearchable https://spivey.oriel.ox.ac.uk/corner/Obfuscated_PDF

tl;dr create a custom font with the glyphs in different places, substitution cipher style. if you copy what you see, you get the symbol value which is not the glyph on the screen
